### PR TITLE
chore: enable tailwind macros and migrate simple uses

### PR DESCRIPTION
**Pull request by tyleroooo**
After this merges you need to add the tailwind vs code extension and add the following to your user preferences json in vscode:

```
"tailwindCSS.classAttributes": [
    "class",
    "className",
    "ngClass",
    "class:list",
    "tw"
  ],
  "tailwindCSS.experimental.classRegex": [
    "tw`([^`]*)", // tw`...`
    "tw=\"([^\"]*)", // <div tw="..." />
    "tw={\"([^\"}]*)", // <div tw={"..."} />
    "tw\\.\\w+`([^`]*)", // tw.xxx`...`
    "tw\\(.*?\\)`([^`]*)" // tw(Component)`...`
  ]
```

and if you like emmet:
```
"tailwindCSS.emmetCompletions": true,
  "emmet.includeLanguages": {
    "\"typescript\"": "\"typescriptreact\""
  },
  "emmet.syntaxProfiles": {
    "jsx": {
      "markup.attributes": {
        "class": "tw",
        "class*": "tw"
      },
      "self_closing_tag": true
    }
  }
```

Note I purposefully am not importing twin.macro's global styles because it breaks stuff in our styles in very subtle ways.

### CODE REVIEW

File: src/pages/markets/NewMarket.tsx [hunk 1]
```diff
@@ -165,18 +169,6 @@ const $ContentSectionHeader = styled(ContentSectionHeader)`
     }
   }
 `;
-
-const $HeaderSection = styled.section`
-  ${layoutMixins.contentSectionDetached}
```
Comment (tyleroooo) [inline src/pages/markets/NewMarket.tsx:170]: bug: missed this

File: tailwind.config.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      'layer-7': 'var(--color-layer-7)',
+
+      border: 'var(--color-border)',
+      'border-white': 'var(--color-border-white)',
+      'border-red': 'var(--color-border-red)',
+
+      'text-0': 'var(--color-text-0)',
+      'text-1': 'var(--color-text-1)',
+      'text-2': 'var(--color-text-2)',
+      'text-button': 'var(--color-text-button)',
```
Comment (jaredvu) [inline tailwind.config.js:41]: textColor would be more descript, so usage becomes `text-textColor-0` instead of `text-text-0`

File: src/App.tsx [hunk 3]
```diff
@@ -130,7 +130,7 @@ const Content = () => {
 
         {isTablet ? <FooterMobile /> : <FooterDesktop />}
 
-        <$NotificationsToastArea />
+        <NotificationsToastArea tw="z-[2] [grid-area:Main]" />
```
Comment (jaredvu) [inline src/App.tsx:133]: `[grid-area:Main]` seems weird. I thought `[]` was for property value only. Does TW not have a grid-area classname? `grid-area-[Main]`?

**Review by jaredvu — APPROVED** [review]

File: src/App.tsx (hunk 3 above)
Comment (tyleroooo) [reply src/App.tsx:133]: Yeah no tailwind shortcuts for it. I think it's still pretty readable here though.

File: tailwind.config.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    screens: {
+      mobile: { max: BREAKPOINT_REM.mobile },
+      notMobile: { min: BREAKPOINT_REM.mobile },
+      tablet: { max: BREAKPOINT_REM.tablet },
+      notTablet: { min: BREAKPOINT_REM.tablet },
+      desktopSmall: { max: BREAKPOINT_REM.desktopSmall },
+      desktopMedium: { min: BREAKPOINT_REM.desktopMedium },
+      desktopLarge: { min: BREAKPOINT_REM.desktopLarge },
+    },
+    colors: {
```
Comment (moo-onthelawn) [inline tailwind.config.js:17]: oh this is nifty

Comment (tyleroooo) [reply tailwind.config.js:17]: yeah this is what allows shortcuts like `text-color-text-0` and `bg-color-layer-4` or just `bg-color-accent`

**Pull request merged**